eoscostarica/eos-rate
# Fix/178 rating

### Rate BP page #178 #179 

- get data from the blockchain (EOS JS API)
- add eosjs package
- create a function to get color using account Name (this way we avoid repeat colors)
- validate user rating data
- handle empty bp objects
- populate rating slider data
- add a message if user rating a bp without login
- add a mutation to save user rating
- add a mutation to update user rating
- add a query to get user rating
- solve a lot of warning message in the console
- clean code
- add soft compare rating functionality in the rate-page radar

### Steps to test

_First scenario_ 
1. login on EOS RATE using scatter
2. go to block producer pages
3. choose one BP and click it
4. if you rating it in the past the slider component would be populated with your last rating values if is your first time all sliders should have initial value (0)
5. change the sliders values
6. click Publish Rating button (your rating should be saved or updated)

_Second scenario_ 
1. using EOS RATE without login
2. go to block producer pages
3. choose one BP and click it
4. All sliders should have initial value (0)
5. change the sliders values
6. click Publish Rating button
7. a warning message should be shown saying that you need login if you want to rate a BP
8. login using scatter
9. click again Publish Rating button

### Checklist
